tests/tests.rs

```diff
@@ -229,50 +229,79 @@ async fn test_on_disk_index() -> Result<()> {
     common::clean(storage, path).await
 }
 
-#[tokio::test]
-async fn test_work_dir_lock() {
-    use nix::sys::wait::{waitpid, WaitStatus};
-    use nix::unistd::{fork, ForkResult};
-
-    let path = common::init("work_dir_lock");
-
+#[test]
+fn test_work_dir_lock() {
+    use rusty_fork::{fork, rusty_fork_id, rusty_fork_test_name};
+    use std::fs::{create_dir_all, read, write};
+    use std::panic;
+    use std::path::Path;
+    use std::sync::Arc;
+    use tokio::runtime::Builder;
+
+    const MAX_CHILD_WAIT_COUNT: usize = 30;
+    const MAX_PARENT_WAIT_SECONDS: u64 = 120;
+
+    let path = Arc::new(common::init("test_work_dir_lock"));
+    let init_file = Path::new(path.as_ref()).join("main_init_ready.special");
+    let child_init_file = init_file.clone();
+    let child_path = path.clone();
     // We need separate processes because locks do not work within the same process
-    match unsafe { fork() } {
-        Ok(ForkResult::Parent { child }) => {
+    fork(
+        rusty_fork_test_name!(test_work_dir_lock),
```

Comment:
According to my tests, error in the function name does not lead to the test failure. I suggest to add explicit check, that child process was started. This can be done by creating a file on the disk right at the beginning of the child process (the same way, like you create file to wait for initialization complete), and check that that file exists at the end of parent process